datadog_checks_base/datadog_checks/base/checks/base.py

```diff
@@ -388,6 +389,16 @@ def http(self):
 
         return self._http
 
+    @property
```

Comment:
nit - why not `@cached_property` without the if statement?

Reply:
Python 2.